winget-install.ps1

```diff
@@ -156,24 +156,48 @@ function Add-ScopeMachine {
 }
 
 #Check if app is installed
-function Confirm-Install ($AppID) {
-    #Get "Winget List AppID"
-    $InstalledApp = & $winget list --Id $AppID -e --accept-source-agreements | Out-String
+function Confirm-Install ($AppID, $AppArgs) {
+    #Get "Winget List AppID" and remove any header information
+    $InstalledApp = & $winget list --Id $AppID -e --accept-source-agreements
+    $InstalledApp = $InstalledApp[$InstalledApp.Length-1]
 
     #Return if AppID exists in the list
-    if ($InstalledApp -match [regex]::Escape($AppID)) {
-        return $true
-    }
-    else {
-        return $false
+    if ($InstalledApp -notmatch [regex]::Escape($AppID)) {
+       return $false 
+    }
+    #Get installed app version
+    if (-not ([string]::IsNullOrEmpty($AppArgs)) ) {
+        & $winget export -s winget --include-versions -o $env:TEMP\temp.json
+        $json = Get-Content -Path $env:TEMP\temp.json | ConvertFrom-Json
+        $Packages = $json.Sources[0].Packages
+        $InstalledVersion = $Packages | Where-Object {$_.PackageIdentifier -eq $AppID} | Select-Object -ExpandProperty Version
+    }
+    #Convert the version variables to correct format (22.01 -> 22.01.0.0 (Major.Minor.Build.Revision)) otherwise comparing values sometimes does not work correct due to wrong format
+    if (-not ([string]::IsNullOrEmpty($AppArgs)) -and $AppArgs -match [regex]::Escape("--version") ) {
+        $AppArgs = $AppArgs.Split(" ")
+        $wordIndex = $AppArgs.IndexOf("--version")
+        [String]$AppVersion = $AppArgs[$wordIndex + 1]
+        while ($InstalledVersion.Split(".").count -ne 4) {
+            $InstalledVersion += ".0"
+        }
+        while ($AppVersion.Split(".").count -ne 4) {
+            $AppVersion += ".0"
+        }
+        if ([Version]$InstalledVersion -ge [Version]$AppVersion) {

```

Comment:
We cannot really compare version like this because sometimes it happens that some apps use SemVer (see https://semver.org/) and it might be incompatible with [version] from PowerShell :/

Reply:
Could you give me an example what you mean and where this would be a problem?

Reply:
I mean, why shouldn´t it be compatible with SemVer? All of the packages use either Major.Minor, Major.Minor.Patch or Major.Minor.Patch.Revision and i made 100% it is compatible

Reply:
Ah, ok. Nvm, i didnt see some rarely use something like Major.Minor.Patch-Revision or something or similar things, but tbh, its difficult to compare version numbers properly in powershell, this was the safest way i could reliable determine the correct version, but you are correct, those edge cases should be covered aswell. 
I will look into it, thanks for bringing it to my attention

Reply:
Maybe i could try to implement a simpler version of this https://github.com/chiprunner1995/PoshSemanticVersion

I will give this a try